service/config.go

```diff
@@ -336,8 +336,20 @@ func (c *NitroClient) DeleteResource(resourceType string, resourceName string) e
 //DeleteResourceWithArgs deletes a resource of supplied type and name. Args are supplied as an array of strings
 //Each array entry is formatted as "key:value"
 func (c *NitroClient) DeleteResourceWithArgs(resourceType string, resourceName string, args []string) error {
-
-	_, err := c.listResourceWithArgs(resourceType, resourceName, args)
+	//Remove unwanted argument (username) for listing but keep it for delete operation
+	var err error
+	if resourceType == "snmptrap_snmpuser_binding" {
+		get_args := make([]string, 0)
+		for i, v := range args {
+			if !strings.Contains(v, "username:") {
+				//append all elements except username 
```

Comment:
This comment is not required. Your code explains it.
Maybe you can explain the intention behind it. Not what your code is doing.